openlibrary/catalog/add_book/match.py

```diff
@@ -1,34 +1,29 @@
+import re
+import unicodedata
 import web
-from deprecated import deprecated
-from openlibrary.catalog.utils import expand_record
-from openlibrary.catalog.merge.merge_marc import editions_match as threshold_match
-
 
 threshold = 875
 
+# fields needed for matching:
+# title, subtitle, isbn, publish_country, lccn, publishers, publish_date, number_of_pages, authors
 
-def db_name(a):
-    date = None
-    if a.birth_date or a.death_date:
-        date = a.get('birth_date', '') + '-' + a.get('death_date', '')
-    elif a.date:
-        date = a.date
-    return ' '.join([a['name'], date]) if date else a['name']
-
+re_amazon_title_paren = re.compile(r'^(.*) \([^)]+?\)$')
+re_brackets = re.compile(r'^(.+)\[.*?\]$')
+# re_brace = re.compile('{[^{}]+?}')
+re_normalize = re.compile('[^[:alpha:] ]', re.I)
+re_whitespace_and_punct = re.compile(r'[-\s,;:.]+')
 
-@deprecated('Use editions_match(candidate, existing) instead.')
-def try_merge(candidate, edition_key, existing):
-    return editions_match(candidate, existing)
+isbn_match = 85
```

Comment:
Using upper case for constants would make the code more readable.

The regexes need to be extended for international punctuation (e.g. French guillemot, Spanish upside down question mark, etc) or better yet, switch to using appropriate Unicode character classes rather than hardcoded lists. Using the [regex](https://pypi.org/project/regex/) module would provide more powerful functionality.

Reply:
Constants upper cased.